src/newrelic.ts

```diff
@@ -40,53 +60,367 @@ function addSegment(spec: Spec) {
+                if(peract) {
+                    (peract.endSegment())
+                }
+            }
+        },
+        extractFromSpec: () => {
+
+        }
+    }
 }
 
+function preload(this: any, opts: any) {
+    const seneca = this;
+    const { options }: { options: NewRelicOptions } = opts;
+    const segmentIsEnabled = options && options.segment && options.segment.ENABLED;
+    const tracingIsEnabled = options && options.tracing && options.tracing.ENABLED;
+    const metricsIsEnabled = options && options.metrics && options.metrics.ENABLED;
+    
+    let telemetryCollector: Nullable<ReturnType<typeof TelemetryCollector>> = null;
+    if (tracingIsEnabled && options.tracing) {
+        const { ACCOUNT_API_KEY, SERVICE_NAME, HOST } = options.tracing;
+        telemetryCollector = TelemetryCollector(
+            ACCOUNT_API_KEY, SERVICE_NAME, HOST
+        );
+    }
+
+    seneca.order.inward.add((spec: Spec) => {
+        if (segmentIsEnabled) {
+            changeSpec(spec, 'inward').addSegments();
+        }
+        if (telemetryCollector) {
+            const specMetadata = telemetryCollector.extractFromSpec(spec, 'inward');
+            telemetryCollector.updateSpecList(specMetadata);
+        }
+    })
+
+    seneca.order.outward.add((spec: Spec) => {
+        if (segmentIsEnabled) {
+            changeSpec(spec, 'outward').endSegments()
+        }
+        if (telemetryCollector) {
+            const specMetadata = telemetryCollector.extractFromSpec(spec, 'outward');
+            telemetryCollector.updateSpecList(specMetadata);
+        }
+    })
 
-function newrelic(this: any, _options: any) {
-  // const seneca: any = this
+    if (metricsIsEnabled) {
+      if (!options.metrics?.ACCOUNT_API_KEY) {
+        throw new Error("Please provide ACCOUNT_API_KEY parameter to Metrics API");
+      }
+      const metricsClient = new MetricClient({
+        apiKey: options.metrics.ACCOUNT_API_KEY,
+      })
+  
+      this.metricsClient = metricsClient;
+    }
 }
 
+function newrelic(this: any, options: NewRelicOptions) {
+    const seneca: any = this
+
+    seneca
+        .message('plugin:newrelic,get:info', get_info)
+        .message('plugin:newrelic,api:metrics', metric_handler)
+        .message('plugin:newrelic,api:metrics', metrics_validation)
+
+    async function get_info(this: any, _msg: any) {
+        return {
+            ok: true,
+            name: 'newrelic',
+            details: {
+                sdk: 'newrelic'
+            }
+        }
+    }
+
+    const _metricBatchBuilder = (metric: MetricBase<any>) => {
+      const batch = new MetricBatch({}, Date.now(), 1000);
+      batch.addMetric(metric)
+      return batch;
+    };
+
+    const _sendBatch = (batch: MetricBatch) => {
+      seneca.metricsClient.send(batch, (err: any, res: any, body: any) => {
+      if (err) {
+          console.log('aq', err);
+      }
+          console.log(res.statusCode);
```

Comment:
Is the `res.statusCode` guaranteed to _not_ be a 4** or 5** code? If it's not guaranteed, then should we handle it?

Reply:
Any suggestions to handle this @lilsweetcaligula ?!
Maybe we can have a log file to write down failed operations